boards/weact_f401ccu6/src/main.rs

```diff
@@ -0,0 +1,459 @@
+//! Board file for WeAct STM32F401CCU6 Core Board
+//!
+//! - <https://github.com/WeActTC/MiniF4-STM32F4x1>
+
+#![no_std]
+// Disable this attribute when documenting, as a workaround for
+// https://github.com/rust-lang/rust/issues/62184.
+#![cfg_attr(not(doc), no_main)]
+#![deny(missing_docs)]
+
+use capsules::virtual_alarm::VirtualMuxAlarm;
+use components::gpio::GpioComponent;
+use kernel::capabilities;
+use kernel::common::dynamic_deferred_call::{DynamicDeferredCall, DynamicDeferredCallClientState};
+use kernel::component::Component;
+use kernel::hil::led::LedLow;
+use kernel::Platform;
+use kernel::{create_capability, debug, static_init};
+
+use stm32f401cc::interrupt_service::Stm32f401ccDefaultPeripherals;
+
+/// Support routines for debugging I/O.
+pub mod io;
+
+// Unit tests
+#[allow(dead_code)]
+mod multi_alarm_test;
+
+// Number of concurrent processes this platform supports.
+const NUM_PROCS: usize = 4;
+
+// Actual memory for holding the active process structures.
+static mut PROCESSES: [Option<&'static dyn kernel::procs::ProcessType>; NUM_PROCS] =
+    [None, None, None, None];
+
+static mut CHIP: Option<&'static stm32f401cc::chip::Stm32f4xx<Stm32f401ccDefaultPeripherals>> =
+    None;
+
+// How should the kernel respond when a process faults.
+const FAULT_RESPONSE: kernel::procs::FaultResponse = kernel::procs::FaultResponse::Panic;
+
+/// Dummy buffer that causes the linker to reserve enough space for the stack.
+#[no_mangle]
+#[link_section = ".stack_buffer"]
+pub static mut STACK_MEMORY: [u8; 0x2000] = [0; 0x2000];
```

Comment:
My new peripheral approach unfortunately uses a large amount of stack on the stm32 chips because of how static_init works -- by allocating all the peripherals in a single call, all of the peripherals are temporarily placed on the stack at once, which is actually more than 4kB. Until I submit a fix for this, `0x2000` is necessary for the stack.

Reply:
Ah, that makes sense then. I'll mark this as resolved, although maybe it's worth adding a comment or linking to the issue?

Reply:
Once I have a fix I will update all the stack sizes back down to where they were in the PR